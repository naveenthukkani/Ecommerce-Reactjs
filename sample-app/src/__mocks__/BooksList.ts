import {Books} from "../../src/home/Books";

export const BooksList: Array<Books> = [
   {
       name:"maths",
       discription: "it is simple problems book for children",
       author:"naveen",
       price: 5,
       pagecount: 1,
	   isbn:"",
	   id:1,
   },
   {
    name:"social",
    discription: "it is sample book for children",
    author:"naveen",
    price: 7,
    pagecount: 1,
	isbn:"",
	id:2,
},
{
    name:"telugu",
    discription: "it is language learning book",
    author:"naveen",
    price: 8,
    pagecount: 1,
	isbn:"",
	id:3,
},
{
    name:"English",
    discription: "it is international language learning book",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:4,
},
{
    name:"maths",
    discription: "it is simple problems book for children",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:5,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:6,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:7,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:8,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:9,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:10,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:11,
}
]