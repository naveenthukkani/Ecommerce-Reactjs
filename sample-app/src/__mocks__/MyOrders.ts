import {Orders} from "../../src/myorders/Orders";

export const OrdersList: Array<Orders> = [
        {
            date: new Date().toLocaleString(),
            status: "Delivered",
            isbn: "9781593275846",
            title: "Eloquent JavaScript, Second Edition",
            subtitle: "A Modern Introduction to Programming",
            author: "Marijn Haverbeke",
            pages: 472,
            price: 500,
            id:1
        },
];