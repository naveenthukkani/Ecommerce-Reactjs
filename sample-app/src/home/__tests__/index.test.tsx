import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {Home} from '../index';
import { BooksList } from '../../__mocks__/BooksList';

describe('Home', () =>{
    let props:any;
    beforeEach(()=>{
        props = {
         listOfBooks: BooksList,
         actions: {
            fetchBooksList: jest.fn()
         }
        }
    });

    it('renders the component correctly', () => {
        const component = shallow(<Home {...props} />);
        expect(toJson(component)).toMatchSnapshot();
      });
});




