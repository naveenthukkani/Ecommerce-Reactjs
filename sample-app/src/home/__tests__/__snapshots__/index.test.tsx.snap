// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the component correctly 1`] = `
<div
  className="Books-list"
>
  <div
    className="Book"
    key="1"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Eloquent JavaScript, Second Edition
    </p>
    <p
      className="Book-Description"
    >
      JavaScript lies at the heart of almost every modern web application, from social apps to the newest browser-based games. Though simple for beginners to pick up and play with, JavaScript is a flexible, complex language that you can use to build full-scale applications.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="2"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Learning JavaScript Design Patterns
    </p>
    <p
      className="Book-Description"
    >
      With Learning JavaScript Design Patterns, you'll learn how to write beautiful, structured, and maintainable JavaScript by applying classical and modern design patterns to the language. If you want to keep your code efficient, more manageable, and up-to-date with the latest best practices, this book is for you.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="3"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Speaking JavaScript
    </p>
    <p
      className="Book-Description"
    >
      Like it or not, JavaScript is everywhere these days-from browser to server to mobile-and now you, too, need to learn the language or dive deeper than you have. This concise book guides you into and through JavaScript, written by a veteran programmer who once found himself in the same position.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="4"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Programming JavaScript Applications
    </p>
    <p
      className="Book-Description"
    >
      Take advantage of JavaScript's power to build robust web-scale or enterprise applications that are easy to extend and maintain. By applying the design patterns outlined in this practical book, experienced JavaScript developers will learn how to write flexible and resilient code that's easier-yes, easier-to work with as your code base grows.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="5"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Understanding ECMAScript 6
    </p>
    <p
      className="Book-Description"
    >
      ECMAScript 6 represents the biggest update to the core of JavaScript in the history of the language. In Understanding ECMAScript 6, expert developer Nicholas C. Zakas provides a complete guide to the object types, syntax, and other exciting changes that ECMAScript 6 brings to JavaScript.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="6"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      You Don't Know JS
    </p>
    <p
      className="Book-Description"
    >
      No matter how much experience you have with JavaScript, odds are you don’t fully understand the language. As part of the 
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="7"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Git Pocket Guide
    </p>
    <p
      className="Book-Description"
    >
      This pocket guide is the perfect on-the-job companion to Git, the distributed version control system. It provides a compact, readable introduction to Git for new users, as well as a reference to common commands and procedures for those of you with Git experience.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
  <div
    className="Book"
    key="8"
    onClick={[Function]}
  >
    <img
      alt="Book"
      src="book.jpg"
    />
    <p
      className="Book-Description"
    >
      Designing Evolvable Web APIs with ASP.NET
    </p>
    <p
      className="Book-Description"
    >
      Design and build Web APIs for a broad range of clients—including browsers and mobile devices—that can adapt to change over time. This practical, hands-on guide takes you through the theory and tools you need to build evolvable HTTP services with Microsoft’s ASP.NET Web API framework. In the process, you’ll learn how design and implement a real-world Web API.
    </p>
    <button
      className="Buy-button"
      onClick={[Function]}
    >
      Buy Button
    </button>
  </div>
</div>
`;
